docs/observability/tools/OpenTelemetry.md

```diff
@@ -115,6 +115,16 @@ Either way, instrumenting your code with OpenTelemetry seems the right approach
 
 Use the [Azure OpenTelemetry Tracing plugin library for Java](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/core/azure-core-tracing-opentelemetry) to enable distributed tracing across Azure components through OpenTelemetry.
 
+### Manual trace context propagation
+
+The trace context is stored in Thread-local storage. When the application flow involves multiple threads (eg. multithreaded work-queue, asynchronous processing with threads handover) then the traces won't get combined into one end-to-end trace chain with automatic [context propagation](https://opentelemetry.io/docs/concepts/signals/traces/#context-propagation).
+To achieve that you need to manually propagate the trace context ([example in Java](https://opentelemetry.io/docs/instrumentation/java/manual/#context-propagation)) by storing the [trace headers](https://www.w3.org/TR/trace-context/#traceparent-header) along with the work-queue item.
+
+### Telemetry testing
+
+Mission critical telemetry data should be covered by testing. Testing the telemetry is possible when using OpenTelemetry Agent. In automated testing environment he OpenTelemetry Agent can be configured to use [OTLP exporter](https://opentelemetry.io/docs/reference/specification/protocol/exporter/) and point the [OTLP exporter endpoint]((https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#otlp-exporter-span-metric-and-log-exporters)
```

Comment:
I'm also a bit confused about what is the "OpenTelemetry agent" :)

Two things are called "agent" in the docs: 
1. [metrics collector](https://opentelemetry.io/docs/collector/getting-started/) that runs as a separate process 
2. [Java agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation#about) JAR that attaches to the main process and sends data to the collector above.

If it's only the latter, then would it make more sense to move this section to the [Java page](https://opentelemetry.io/docs/instrumentation/java/manual/)?

Reply:
For the agent part: good point, these names are mixed. Going with your understanding I think there is a mistake in this section in the playbook: https://github.com/microsoft/code-with-engineering-playbook/blob/main/docs/observability/tools/OpenTelemetry.md#open-telemetry-implementation-patterns
As the automatic instrumentation is not provided by the Opentelemetry Agent.
Taking a closer look into that

Reply:
The JVM agent is also a kind of collector per definition "OpenTelemetry Collector offers a vendor-agnostic implementation on how to receive, process, and export telemetry data"

Perhaps we could also submit a change request to https://opentelemetry.io/docs/concepts/data-collection/ to reflect the fact that the JVM agent is not "A Collector instance running with the application or on the same host as the application"

Reply:
@algattik I see your point of view, but I can’t find anything in the documentation to back the understanding java agent as a collector. 
Java agent seems to allow for exporting the telemetry but not processing it, so it’s more an automatic instrumentation, not a collector.
The OpenTelemetry Collector term seems to refer to the specific OpenTelemetry project: 
https://opentelemetry.io/docs/collector/#status-and-releases
https://opentelemetry.io/docs/concepts/data-collection/#repositories
There is also a way to build custom collector but the doc doesn’t mention that the OpenTelemetry Java agent implements the Collector contract.
https://opentelemetry.io/docs/collector/custom-collector/

Reply:
The Java agent does have a processor https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#batch-span-processor

Reply:
I see, but still it seems that java agent is more of an alternative to using SDK and not a collector in the OpenTelemetry nomenclature, as the doc says:
> This artifact implements environment-based autoconfiguration of the OpenTelemetry SDK. This can be an alternative to programmatic configuration using the normal SDK builders.

It seems to me that calling java agent a collector may be confusing to readers because it is not consistent with the OpenTelemetry documentation, even though conceptually java agent provides some collector capabilities.